stbt-power

```diff
@@ -1,183 +1,53 @@
-
-    echo "$output" | grep -q BADPARAM &&
-        die "invalid outlet '$outlet' (hint: use the 'status' command)"
-
-    # Prettify the output from the device
-    echo "$output" |
-    sed 's|</*html>||g' |
-    tr "," "\n" |
-    sed -e 's|=1| = ON|g' -e 's|=0| = OFF|g' |
-    ipp_filteroutlet "$outlet" |
-    head -4  # IP Power device prints 8 outlet names, but only has 4.
-}
-ipp_command() {
-    local command=$1 outlet="$2"
-    case "$1" in
-        on) echo "SetPower+$outlet=1";;
-        off) echo "SetPower+$outlet=0";;
-        status) echo "GetPower";;
-    esac
-}
-ipp_filteroutlet() {
-    local outlet="$1"
-    if [ -z "$outlet" ]; then
-        cat
-    else
-        grep "$outlet" ||
-            die "invalid outlet '$outlet' (hint: use the 'status' command)"
-    fi
-}
-
-pdu() {
-    local command=$1 hostname=$2 outlet="$3" pdu_error command_ids data _
-
-    pdu_error="Failed to execute command '$command' on PDU '$hostname' "
-    pdu_error+="outlet '${outlet:-all}'"
-
-    if [[ $command == status ]]; then
-        if [[ -n $outlet ]]; then
-            pdu_status $hostname "$outlet" || die "$pdu_error"
-        else
-            { pdu_status $hostname '1-A[0-9]+' || die "$pdu_error"; } | \
-                awk '{ printf("1-A%d: %s\n", NR, $0) }'  # Add outlet addresses
-        fi
-    else
-        declare -A command_ids=([on]=1 [off]=2)
-
-        # 2 attempts to minimise failure rate
-        for _ in 1 2; do
-            data="selOutlet=%3F$(pdu_outlet_no "$outlet")&"
-            data+="ctrl_all=${command_ids[$command]}"
-            _curl -d "$data" \
-                http://admin:admin@$hostname/Forms/index_3 >/dev/null
-
-            # The above HTTP POST always succeeds, but doesn't always switch
-            # the power outlet. Check by querying the new outlet status,
-            # allowing up to 8 seconds for it to take effect.
-            for _ in {1..8}; do
-                sleep 1
-                pdu_status $hostname "$outlet" | grep -iq "$command" && return
-            done
-        done
-
-        die "$pdu_error"
-    fi
-}
-pdu_status() {
-    local hostname=$1 outlet="$2"
-    _curl http://admin:admin@$hostname | awk "/$outlet/,/<\/ul>/" | \
-        grep -Eo 'ON|OFF'
-}
-pdu_outlet_no() {
-    echo $1 | sed 's/^1-A//'
-}
-_curl() {
-    # 2 attempts to minimise failure rate
-    curl --anyauth --fail --silent "$@" ||
-    curl --anyauth --fail --show-error --silent "$@"
-}
-
-main "$@"
+#!/usr/bin/python
+
+import argparse
+import sys
+from textwrap import dedent
+
+from _stbt.config import get_config
+from _stbt.power import uri_to_pdu
+
+
+def main(argv):
+    parser = argparse.ArgumentParser(
+        description="Send commands to a computer controllable power switch",
+        formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument(
+        "--power-outlet", metavar="URI",
+        default=get_config("global", "power", ""), help=dedent("""\
```

Comment:
This should be `power_outlet`, which is what the help message says and what the previous version used.